src/pe/export.rs

```diff
@@ -56,52 +56,79 @@ pub type ExportOrdinalTable = Vec<u16>;
 #[derive(Debug, Default)]
 /// Export data contains the `dll` name which other libraries can import symbols by (two-level namespace), as well as other important indexing data allowing symbol lookups
 pub struct ExportData<'a> {
-    pub name: &'a str,
+    pub name: Option<&'a str>,
     pub export_directory_table: ExportDirectoryTable,
-    pub export_name_pointer_table: ExportNamePointerTable,
-    pub export_ordinal_table: ExportOrdinalTable,
-    pub export_address_table: ExportAddressTable,
+    pub export_name_pointer_table: Option<ExportNamePointerTable>,
+    pub export_ordinal_table: Option<ExportOrdinalTable>,
```

Comment:
I would say ditto for `export_name_pointer_table` too. I would only use `Option` for these if you can come up with a reason why you want to distinguish between `None` and `Some(vec![])`, ie between invalid RVA and zero size. Otherwise it's simpler for consumers if it's plain `Vec` that is empty for both invalid RVA and zero size.

Reply:
Many thanks to @m4b  and @philipc for the encouraging response.

Yes, my argument for `Option` is to distinguish the case where a [correct offset](https://github.com/m4b/goblin/blob/master/src/pe/export.rs#L77) cannot be obtained (i.e. the table does not exist), from the case where the table's offset can be obtained (i.e. the table exists) but its size is zero.

For executable `PE`(i.e. `.exe` files), I think that the Windows's PE loader will not care much if the export-related tables are valid or not (it is possible to make a runable `.exe` having invalid `export_name_pointer_table`). So these tables "can be skipped".

But I am happy also if we choose to unify these cases :), I don't know if this choice is better or not but, as you have said, it will not break the API and maybe simpler for the consumers.

I have changed the PR to restore these tables to not optional.